datumaro/plugins/celeba_format.py

```
@@ -0,0 +1,151 @@
+        with open(path, encoding='utf-8') as f:
+            for line in f:
+                item_id, item_ann = self.split_annotation(line)
+                label_ids = [int(id) for id in item_ann]
+                anno = []
+                for label in label_ids:
+                    if label > len(label_categories):
+                        for i in range(len(label_categories), label + 1):
+                            label_categories.add('label_%d' % i)
+                    anno.append(Label(label))
+
+                items[item_id] = DatasetItem(id=item_id, image=images.get(item_id),
+                    annotations=anno)
+
+        bbox_path = osp.join(self._anno_dir, CelebaPath.BBOXES_FILE)
+        if osp.isfile(bbox_path):
+            with open(bbox_path, encoding='utf-8') as f:
+                for line in f:
+                    if len(line.split()) < 5 or 'image_id' == line[0:8]:
+                        continue
+                    item_id, item_ann = self.split_annotation(line)
+                    bbox = [float(id) for id in item_ann]
+                    if item_id not in items:
+                        items[item_id] = DatasetItem(id=item_id, image=images.get(item_id))
+                    anno = items[item_id].annotations
+                    label = None
+                    if 0 < len(anno):
+                        label = anno[0].label
+                    anno.append(Bbox(bbox[0], bbox[1], bbox[2], bbox[3], label=label))
+
+        attr_path = osp.join(self._anno_dir, CelebaPath.ATTRS_FILE)
+        if osp.isfile(attr_path):
+            with open(attr_path, encoding='utf-8') as f:
+                f.readline()
+                line = f.readline()
+                name_attr = line.split()
+                attr = {}
+                for line in f:
+                    item_id, item_ann = self.split_annotation(line)
+                    if len(name_attr) != len(item_ann):
+                        continue
```

Continuing here seems questionable; this is an error in the datafile, so we should raise an exception.